opendatahub-io/modelmesh-serving
# Sync kserve/main (v0.11.1) to odh/main

#### Motivation
Sync kserve/main to odh/main. `v0.11.1` tag points to the latest commit in kserve/modelmesh-serving:main.

#### Testing instructions
1. `git clone git@github.com:opendatahub-io/modelmesh-serving.git`
2. `cd modelmesh-serving`

3. `TAG=fast CONTROLLERNAMESPACE=opendatahub NAMESPACE=modelmesh-serving CUSTOM_IMG=modelmesh-controller=quay.io/opendatahub/modelmesh-controller:pr-229 make deploy-mm-for-odh `

4. Check that the modelmesh controller deployment/pod has the correct image (shown in step 3). If not, patch it: `oc patch deployment modelmesh-controller -n opendatahub --type='json' -p='[{"op": "replace", "path": "/spec/template/spec/containers/0/image", "value": "quay.io/opendatahub/modelmesh-controller:pr-229"}]'`
5. Check that the manager container of modelmesh-controller and of odh-model-controller show no error.
6. `TAG=fast CONTROLLERNAMESPACE=opendatahub NAMESPACE=modelmesh-serving FORCE=true make deploy-fvt-for-odh`
7. Check that modelmesh-controller container, odh-model-controller container and runtime containers show no error. 
8. Clean up: `C_MM_CTRL_KFCTL=true C_MM_TEST=true make cleanup-for-odh`

#### PR checklist

Checklist items below are applicable for development targeted to both fast and stable branches/tags
- [x] FVT tests pass locally
- [ ] If the PR adds a new container image or updates the tag of an existing image (not build within cpaas), is the corresponding change made in live-builder and cpaas-midstream to add/update the image tag in the operator CSV? Link the PRs if applicable

Checklist items below are applicable for development targeted to both fast and stable branches/tags
- [ ] Tested modelmesh serving deployment with odh-manifests and ran odh-manifests-e2e tests locally
